internal/accounts/staking-info.go

```diff
@@ -61,18 +61,61 @@ func stakingInfo(
 		return nil, err
 	}
 
-	return &StakingResult{*staking, *delegation}, nil
+	// get staking infos and delegation infos
+	stakingInfos := flowkit.NewStakingInfoFromValue(*staking)
+	delegationInfos := flowkit.NewStakingInfoFromValue(*delegation)
+
+	// get a set of node ids from all staking infos
+	nodeStakes := make(map[string]cadence.Value)
+	for _, stakingInfo := range stakingInfos {
+		nodeID, ok := stakingInfo["id"]
+		if ok {
+			nodeStakes[nodeIDToString(nodeID)] = nil
+		}
+	}
+
+	chain, err := util.GetAddressNetwork(address)
+	if err != nil {
+		return nil, err
+	}
+	// foreach node id, get the node total stake
+	for nodeID := range nodeStakes {
+		stake, err := services.Accounts.NodeTotalStake(nodeID, chain)
+		if err != nil {
+			return nil, err
+		}
+
+		nodeStakes[nodeID] = *stake
+	}
+
+	// foreach staking info, add the node total stake
+	for _, stakingInfo := range stakingInfos {
+		nodeID, ok := stakingInfo["id"]
+		if ok {
+			stakingInfo["nodeTotalStake"] = nodeStakes[nodeIDToString(nodeID)].(cadence.UFix64)
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I feel like this is a presentation (result building) logic, which could be better if moved to the result object, however the result doesn't have a factory method pattern established so there isn't a place that feels this could be done instead of a method on the struct itself. This is a bit nit point so I don't mind if you feel it should stay like this, but to me the command function itself should only do argument preparation (parsing, validation etc) and then the presentation should be handled in result. But then again a bit nit. Example is here: https://github.com/onflow/flow-cli/blob/janez/staking-stats/internal/transactions/transactions.go#L71 but it's a good point that it could be output format-agnostic which means it should have a common place to do this, but still maybe the common place to do this could be a factory method.

Reply:
agreed, that it is not the best place to do this. I decided to move this logic inside `accounts.StakingInfo` I feel like this is better. What do you think.